src/main/java/com/xjeffrose/chicago/server/DBRouter.java

```diff
@@ -1,206 +1,64 @@
 package com.xjeffrose.chicago.server;
 
 import com.xjeffrose.chicago.db.DBManager;
+import com.xjeffrose.chicago.db.RocksDBImpl;
+import com.xjeffrose.chicago.db.StorageProvider;
 import com.xjeffrose.xio.application.Application;
 import com.xjeffrose.xio.bootstrap.ApplicationBootstrap;
 
 import com.xjeffrose.xio.pipeline.XioSslHttp1_1Pipeline;
 import io.netty.channel.ChannelHandler;
+import io.netty.util.internal.PlatformDependent;
 import java.io.Closeable;
 import java.io.IOException;
 import java.net.InetSocketAddress;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class DBRouter implements Closeable {
   private static final Logger log = LoggerFactory.getLogger(DBRouter.class);
 
   //TODO(JR): Make this concurrent to applow for parallel streams
-  private final ChiConfig config;
-  private final RocksDBImpl rocksDbImpl;
+  private final StorageProvider db;
   private final DBManager manager;
-  private final DBLog dbLog;
-  //private ChicagoMasterManager masterManager;
   private final ChannelHandler handler;
-  private Application application;
-
-  public DBRouter(ChiConfig config, RocksDBImpl rocksDbImpl, DBLog dbLog) {
-    this.config = config;
-    this.rocksDbImpl = rocksDbImpl;
-    manager = new DBManager(rocksDbImpl);
-    this.dbLog = dbLog;
-    this.handler = new ChicagoDBHandler(manager, dbLog);
-    //config.setDbRouter(this);
-  }
-  /*
-  private void configureDBServer() {
-    XioServerDef dbServer = new XioServerDefBuilder()
-        .name("Chicago DB Server")
-        .listen(new InetSocketAddress(config.getDBBindIP(), config.getDBPort()))
-//        .withSecurityFactory(new XioNoOpSecurityFactory())
-        .withSecurityFactory(new XioSecurityFactory() {
-          @Override
-          public XioSecurityHandlers getSecurityHandlers(XioServerDef xioServerDef, XioServerConfig xioServerConfig) {
-            return new XioSecurityHandlerImpl(config.getCert(), config.getKey());
-          }
+  private final ChicagoPaxosHandler chicagoPaxosHandler;
```

Comment:
I think the class is the correct place for this, but the name of the class is probably misleading.